models/demos/segformer/tt/ttnn_segformer_efficient_selfattention.py

```diff
@@ -116,42 +94,37 @@ def __call__(
             dtype=ttnn.bfloat8_b,
         )
 
-        # print("Q1", query.shape)
-        hidden_states = ttnn.to_memory_config(hidden_states, ttnn.L1_MEMORY_CONFIG, dtype=ttnn.bfloat16)
-        query = ttnn.to_memory_config(query, ttnn.L1_MEMORY_CONFIG, dtype=ttnn.bfloat16)
+        query = ttnn.to_memory_config(query, ttnn.L1_MEMORY_CONFIG, dtype=ttnn.bfloat8_b)
         # Split Heads
         if self.num_attention_heads == 1:
             query_layer = query
         else:
-            query_layer = self.transpose_for_scores(query)
-
-        # print("Q2", query_layer.shape)
+            query_layer = ttnn.experimental.nlp_create_qkv_heads_segformer(query, memory_config=ttnn.L1_MEMORY_CONFIG)[
+                0
+            ]
 
-        # print("sr0", hidden_states.shape)
         if self.sr_ratio > 1:
             if len(hidden_states.shape) == 3:
                 batch_size, seq_len, num_channels = hidden_states.shape
             elif len(hidden_states.shape) == 4:
                 batch_size, __, seq_len, num_channels = hidden_states.shape
 
+            hidden_states = ttnn.to_memory_config(hidden_states, ttnn.L1_MEMORY_CONFIG, dtype=ttnn.bfloat8_b)
             hidden_states = ttnn.to_layout(hidden_states, layout=ttnn.ROW_MAJOR_LAYOUT)
             hidden_states = ttnn.reshape(hidden_states, (batch_size, height, width, num_channels))
+            hidden_states = ttnn.to_layout(hidden_states, ttnn.TILE_LAYOUT)
```

Comment:
I assume this block is written this way because reshape does not support what you want to do in TILE? Consider adding a comment to give context to reader.